app/services/solidus_bolt/webhooks/sorter.rb

```diff
@@ -1,22 +1,34 @@
 # frozen_string_literal: true
 
+require 'base64'
+
 module SolidusBolt
   module Webhooks
     class Sorter
-      def self.call(params)
-        new(params).call
+      attr_reader :params, :hash_header
+
+      def self.call(params, hash_header)
+        new(params, hash_header).call
       end
 
-      def initialize(params)
+      def initialize(params, hash_header)
         @params = params
+        @hash_header = hash_header
       end
 
       def call
-        handler&.call(@params)
+        return unless bolt_request?
+
+        handler&.call(params)
       end
 
       private
 
+      def bolt_request?
+        signing_secret = SolidusBolt.config.bolt_signing_secret || ''
+        hash_header == Base64.encode64(OpenSSL::HMAC.digest("SHA256", params.to_json, signing_secret))
+      end
```

Comment:
Usually authentication should be handled by controller/action(e.g. cancan authorization). This seems a work for a `before_action` callback in the controller.